### DISCUSSION TITLE
Issue with flet-pyodide in versions 0.25.0 and 0.25.1 when using local HTTP server

### DISCUSSION THREAD

**Discussion by 7576457** [flet-dev/flet#4466] [Q&A] [answered]
### Question

Hello.

I have a fairly large project that heavily relies on the functionality of flet-pyodide. Up until version 0.25.0, my program was working perfectly. To test my project, I use the following command:  
```python.exe -m http.server 8080 --bind 127.0.0.1 --directory dist```

This command gives me a better understanding of how my site will work in reality compared to the command  
```flet run --web```,  
as I noticed that the results differ between these methods.

This issue occurs in versions 0.25.0 and 0.25.1.  
With version 0.24.1, there are no problems at all.

I’m also unsure what code I should provide since this issue doesn’t seem to be related to how I write my code.  
To reproduce this error, here’s what I do:
1. ```flet publish main.py --assets assets --app-name MOON --app-description MOON```
2. ```python.exe -m http.server 8080 --bind 127.0.0.1 --directory dist```

Can anyone help with this? This is very important to me.

### Error message

```bash
PythonError: Traceback (most recent call last):
  File "/lib/Python311.zip/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1209, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1181, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1152, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/pyodide/main.py", line 1, in <module>
    import flet as ft
ModuleNotFoundError: No module named 'flet'
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by mrkito** [comment]
![Uploading Screenshot 2024-12-01 at 13.37.11.png…]()

**Comment by 7576457** [comment]
Who is able to reproduce this issue?

**Comment by 7576457** [comment]
Hello, @ndonkoHenri. I was wondering if I could improve my question about this issue. I really don’t understand what I’m doing wrong when installing the new version of Flet. How can I make my question better?
Since all I need in my case is to install the new version of Flet.

I also noticed that on PyPI, there’s only a pre-release version of `flet-pyodide` available with the new Flet version. Is this pre-release of `flet-pyodide` included in Flet 0.25.1?

**Reply by FeodorFitsner** [reply]
`flet-pyodide` is no longer required (available). It's `flet` all the way on all platforms. See this note: https://flet.dev/blog/flet-new-packaging-pre-release#new-packages-structure

**Comment by 7576457** [comment]
I read it again very carefully. Please explain it to me like I'm a beginner, just one moment. I thought it would be easier to show with examples. To reproduce the problem I am talking about, you can use the code from the documentation.

Look, here is the code from the documentation:

```python
import flet as ft

def main(page: ft.Page):
    page.title = "Flet counter example"
    page.vertical_alignment = ft.MainAxisAlignment.CENTER

    txt_number = ft.TextField(value="0", text_align=ft.TextAlign.RIGHT, width=100)

    def minus_click(e):
        txt_number.value = str(int(txt_number.value) - 1)
        page.update()

    def plus_click(e):
        txt_number.value = str(int(txt_number.value) + 1)
        page.update()

    page.add(
        ft.Row(
            [
                ft.IconButton(ft.Icons.REMOVE, on_click=minus_click),
                txt_number,
                ft.IconButton(ft.Icons.ADD, on_click=plus_click),
            ],
            alignment=ft.MainAxisAlignment.CENTER,
        )
    )
#flet publish main.py --assets assets --app-name MyApp --app-description "My Static Site"
#python -m http.server 8080 --bind 127.0.0.1 --directory dist
ft.app(main)
```

In version 0.24.1, I ran this app with the following commands:

```bash
flet publish main.py --assets assets --app-name MyApp --app-description "My Static Site"
```
```bash
python -m http.server 8080 --bind 127.0.0.1 --directory dist
```

A static site opened up, and it worked completely in the browser along with Pyodide.

Now, in version 0.25.0 and 0.25.1, I use the same commands, but I get the error I mentioned at the top.

I suspect that I'm just banging my head against a wall. The point is, I'm running my other project using the built-in Python tool. Now this doesn't work.

video example:

**- The result I want and get in version 0.24.1**

https://github.com/user-attachments/assets/5462dea3-8212-49a8-b992-49794c5bb22e

**- The result that does not satisfy me in version 0.25.1**

https://github.com/user-attachments/assets/3778c46e-b693-479d-aaa7-dbbc6a8f6592

**Reply by FeodorFitsner** [reply]
do you have `requirements.txt`?

**Reply by 7576457** [reply]
For this example with the code from the documentation - no

**Reply by FeodorFitsner** [reply, marked as answer]
Add `requirements.txt` with this:

```
flet
```

Re-publish the app. Let me know if that worked.

**Reply by 7576457** [reply]
It worked. Thank you for your hard work and patience!

**Reply by FeodorFitsner** [reply]
Good. But I think it's a "bug" as behavior has changed from 0.24.x.
I think we need to implicitly add `flet` as dependency as no `requirements.txt` found.

**Reply by FeodorFitsner** [reply]
Created a new issue: https://github.com/flet-dev/flet/issues/4493
